### PR TITLE
add: integration with client-cloud-services

### PR DESCRIPTION
**Pull request by nevil-mathew**

### CODE REVIEW

File: src/database/migrations/20240502174600-add_email_templates.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+module.exports = {
+	up: async (queryInterface, Sequelize) => {
+		const defaultOrgId = queryInterface.sequelize.options.defaultOrgId
+
+		if (!defaultOrgId) {
+			throw new Error('Default org ID is undefined. Please make sure it is set in sequelize options.')
+		}
+
+		// Path to the file to be uploaded
+		const fileName = 'emailLogo.png'
```
Comment (rakeshSgr) [inline src/database/migrations/20240502174600-add_email_templates.js:63]: @nevil-mathew we should add this doc, how to change the email logo

File: src/generics/utils.js [hunk 2]
```diff
@@ -46,44 +48,21 @@ const composeEmailBody = (body, params) => {
...
-			bucketName: process.env.DEFAULT_OCI_BUCKET_NAME,
-			endpoint: process.env.OCI_BUCKET_ENDPOINT,
-		}
-		imgPath = await OciFileHelper.getDownloadableUrl(options)
+const getDownloadableUrl = async (filePath) => {
+	let bucketName = process.env.CLOUD_STORAGE_BUCKETNAME
+	let expiryInSeconds = parseInt(process.env.SIGNED_URL_EXPIRY_IN_SECONDS) || 300
+
+	let response = await cloudClient.getSignedUrl(bucketName, filePath, expiryInSeconds, common.READ_ACCESS)
+	console.log(response)
```
Comment (rakeshSgr) [inline src/generics/utils.js:56]: @nevil-mathew remove the console log

File: src/services/files.js [hunk 3]
```diff
@@ -22,26 +22,29 @@ module.exports = class FilesHelper {
...
-			} else if (process.env.CLOUD_STORAGE === 'OCI') {
-				response = await cloudServices.getOciSignedUrl(destFilePath)
+			// decide on which bucket has to be passed based on api call
+			if (isAssetBucket) {
+				cloudBucket = process.env.PUBLIC_ASSET_BUCKETNAME
+			} else {
+				cloudBucket = process.env.CLOUD_STORAGE_BUCKETNAME
 			}
-			response.destFilePath = destFilePath
+			let expiryInSeconds = parseInt(process.env.SIGNED_URL_EXPIRY_IN_SECONDS) || 300
```
Comment (rakeshSgr) [inline src/services/files.js:40]: @nevil-mathew is the expire time will work for all cloud service with seconds ?

File: src/envVariables.js [hunk 4]
```diff
@@ -127,26 +79,6 @@ let enviromentVariables = {
...
-	},
-	OCI_BUCKET_REGION: {
-		message: 'Required oci bucket region',
-		optional: process.env.CLOUD_STORAGE === 'OCI' ? false : true,
-	},
-	OCI_BUCKET_ENDPOINT: {
-		message: 'Required oci bucket endpoint',
-		optional: process.env.CLOUD_STORAGE === 'OCI' ? false : true,
-	},
-	DEFAULT_OCI_BUCKET_NAME: {
```
Comment (rakeshSgr) [inline src/envVariables.js:146]: @nevil-mathew update .env sample and docker files

File: src/envVariables.js [hunk 5]
```diff
@@ -288,16 +220,48 @@ let enviromentVariables = {
...
 		optional: true,
-		default: 3600000,
+		default: 300,
 	},
 	ALLOWED_ACTIVE_SESSIONS: {
 		message: 'Require allowed active sessions',
 		optional: true,
 		default: 0,
 	},
+	CLOUD_STORAGE_PROVIDER: {
```
Comment (rakeshSgr) [inline src/envVariables.js:233]: @nevil-mathew add these in .env-sample files and docker files

File: src/package.json [hunk 6]
```diff
@@ -32,6 +32,7 @@
 		"body-parser": "^1.19.0",
 		"bullmq": "^4.12.3",
 		"cli-table": "^0.3.11",
+		"client-cloud-services": "^8.0.0",
```
Comment (rakeshSgr) [inline src/package.json:35]: @nevil-mathew remove this package elevate-cloud-storage if we are not using

File: src/database/migrations/20240502174600-add_email_templates.js (hunk 1 above)
Comment (nevil-mathew) [reply src/database/migrations/20240502174600-add_email_templates.js:63]: Yes, but we have not made any doc for 3.0 changes, right?

File: src/services/files.js (hunk 3 above)
Comment (nevil-mathew) [reply src/services/files.js:40]: Yes, its updated to handle that

**Pull request merged**